pkg/oc/admin/diagnostics/diagnostics/cluster/app_create/route.go

```diff
@@ -0,0 +1,159 @@
+		d.out.Error("DCluAC022", err, fmt.Sprintf("%s: Creating route '%s' failed:\n%v", now(), d.appName, err))
+		return false
+	}
+	return true
+}
+
+// check that the route is admitted and can be reached
+func (d *AppCreate) testRoute() bool {
+
+	if !d.testRouteAdmitted() {
+		return false
+	}
+
+	if d.skipRouteConnect {
+		d.out.Debug("DCluAC023", "skipping route connection test as requested")
+		return true
+	}
+	return d.testRouteConnection()
+}
+
+func (d *AppCreate) testRouteAdmitted() bool {
+	defer recordTime(&d.result.Route.ReadyTime)
+
+	// set up a watch for route admission
+	d.out.Debug("DCluAC024", fmt.Sprintf("%s: Waiting for route to be admitted by a router", now()))
+	watcher, err := d.RouteClient.Route().Routes(d.project).Watch(metav1.ListOptions{FieldSelector: "metadata.name=" + d.appName, TimeoutSeconds: &d.routeAdmissionTimeout})
+	if err != nil {
+		d.out.Error("DCluAC025", err, fmt.Sprintf(`
+%s: Failed to establish a watch for '%s' route to be admitted by a router:
+  %v
+This may be a transient error. Check the master API logs for anomalies near this time.
+  		`, now(), d.appName, err))
+		return false
+	}
+	defer stopWatcher(watcher)
+
+	// test for the result of the watch
+	for event := range watcher.ResultChan() {
+		ready, err := isRouteAdmitted(event)
+		if err != nil {
+			d.out.Error("DCluAC026", err, fmt.Sprintf(`
+%s: Error while watching for route to be admitted by a router:
+  %v
+This may be a transient error. Check the master API logs for anomalies near this time.
+  			`, now(), err))
+			return false
+		}
+		if ready {
+			d.out.Debug("DCluAC027", fmt.Sprintf("%s: Route has been admitted by a router", now()))
+			return true
+		}
+	}
+	d.out.Error("DCluAC028", nil, fmt.Sprintf(`
+%s: Route was not admitted by a router before timeout (%d sec)
+Diagnostics waited for the '%s' route to be admitted by a router (making the
+application available via that route) after the test app started running.
+However, this did not occur within the timeout.
+Some of the reasons why this may fail include:
+  * There is no router running to accept routes
+  * The available router(s) are configured not to accept the route
+  * The router simply needs longer to admit the route (you can increase the timeout)
+  * The app stopped responding or was killed
+	`, now(), d.routeAdmissionTimeout, d.appName))
+	return false
+}
+
+func isRouteAdmitted(event watch.Event) (bool, error) {
+	switch event.Type {
+	case watch.Deleted:
+		return false, errors.NewNotFound(schema.GroupResource{Resource: "routes"}, "")
+	}
+	switch r := event.Object.(type) {
+	case *route.Route:
+		for _, ingress := range r.Status.Ingress {
+			for _, cond := range ingress.Conditions {
+				if cond.Type == route.RouteAdmitted && cond.Status == "True" {
+					return true, nil
+				}
+			}
+		}
+		return false, nil
+	}
+	return false, nil
+}
+
+// test that we get the expected HTTP response from the route
+func (d *AppCreate) testRouteConnection() bool {
```

Comment:
Maybe I'm a bit picky, but can you rename this to `checkRouteConnection` (the same applies to comment: verify that we get...). The name `test*` in golang usually is used for testing which seems confusing when quickly skimming the code. The same applies to other similar methods for DC, Service.